x-pack/test/functional/services/ml/api.ts

```diff
@@ -250,8 +250,17 @@ export function MachineLearningAPIProvider({ getService }: FtrProviderContext) {
       const { body, status } = await esSupertest
         .put(`/_inference/${taskType}/${inferenceId}`)
         .send(requestBody);
-      this.assertResponseStatusCode(200, status, body);
-
+      if (status === 408) {
```

Comment:
Conditional assertion in a test arrangement code doesn't seem right to me. I believe It should happen in a particular test suite when certain conditions are met.

Reply:
Changed test suite a bit to handle request time out case and request status. https://github.com/elastic/kibana/pull/187521/commits/e95920c44306757da872756e31321179cfdc5e9e